plugin/src/main/java/com/redhat/foreman/ForemanComputerListener.java

```diff
@@ -21,15 +24,22 @@
     private static final Logger LOGGER = Logger.getLogger(ForemanComputerListener.class.getName());
 
     @Override
+    @SuppressFBWarnings(value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
     public void onLaunchFailure(Computer c, TaskListener taskListener) throws IOException, InterruptedException {
         try {
             super.onLaunchFailure(c, taskListener);
         } catch (Exception e) {
-            LOGGER.log(Level.SEVERE, "Uncaught unexpected exception occurred while calling super.onLaunchFailed(): ", e);
+            LOGGER.log(Level.SEVERE,
+                    "Uncaught unexpected exception occurred while calling super.onLaunchFailed(): ", e);
         }
         if (c instanceof ForemanComputer) {
-            LOGGER.info("Launch of the Computer '" + c.getDisplayName() + "' failed, releasing...");
-            ((ForemanComputer) c).eagerlyReturnNodeLater();
+            ForemanComputer fc = (ForemanComputer) c;
+            CloudStatistics.get().attach(
+                    CloudStatistics.get().getActivityFor(fc.getId()),
+                    CloudStatistics.get().getActivityFor(fc.getId()).getCurrentPhase(),
+                    new PhaseExecutionAttachment(ProvisioningActivity.Status.FAIL, c.getLog()));
```

Comment:
Prefix `c.getLog()` with "Provisioning failed with: " otherwise it can be presented to user as a failure with empty message which is confusing.